### PR TITLE
d3.geoIdentity can be used as a function and has an inverse

### PR DESCRIPTION
**Pull request by Fil**
solves https://github.com/d3/d3-geo/issues/182

(maybe?)

### CODE REVIEW

**Comment by Fil** [comment]
@martgnz @LuisSevillano might want to confirm it works for their use case.

All tests are green and a build is @ https://files-kof3v3c1z.now.sh/d3-geo.js

File: src/projection/identity.js [hunk 1]
```diff
@@ -24,39 +24,45 @@ export default function() {
...
-    },
-    fitWidth: function(width, object) {
-      return fitWidth(projection, width, object);
-    },
-    fitHeight: function(height, object) {
-      return fitHeight(projection, height, object);
-    }
+  projection = function(p) {
+    return [p[0] * k + tx, p[1] * k + ty];
+  };
```
Comment (mbostock) [inline src/projection/identity.js]: The projection function is never reassigned, so how about:

```js
function projection(p) {
  return …;
}
```

**Comment by Fil** [comment]
new build at https://files-822dihw5b.now.sh/d3-geo.js

**Comment by martgnz** [comment]
I've tested a bit and it works perfectly @Fil https://observablehq.com/d/0cc849950b7d5b96

**Comment by Fil** [comment]
I had forgotten to check for identity.reflectX/reflectY:

here's a before/after :)

Capture d’écran 2019-12-07 à 17 26 05

https://observablehq.com/d/0b817c9a2552c573

**Comment by Fil** [comment]
I haven't yet merged because I don't like the duplication of the linear transform in [L9](https://github.com/d3/d3-geo/blob/12206c2e4d386602797ef94455f057c47c30dd6d/src/projection/identity.js#L9) and [L27](https://github.com/d3/d3-geo/blob/12206c2e4d386602797ef94455f057c47c30dd6d/src/projection/identity.js#L27)

I have a different version at https://gist.github.com/Fil/25f27c384495f7f8cdf686080e81f7df which centralizes this linear transform in projection(); overall the code ends up being much simpler — with the inconvenient that for the real identity (scale=1, translate=0,0), it does two useless multiplications (&times;1) and additions (+0).

**Comment by Fil** [comment]
superseded by #184

**Pull request closed without merging**